packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -69,222 +74,301 @@ abstract class FlutterGoldenFileComparator extends GoldenFileComparator {
-    return FlutterGoldensRepositoryFileComparator(baseDirectory.uri);
+    goldens ??= SkiaGoldClient(baseDirectory);
+    await goldens.auth();
+    await goldens.imgtestInit();
+    return FlutterSkiaGoldFileComparator(baseDirectory.uri, goldens);
   }
 
   @override
   Future<bool> compare(Uint8List imageBytes, Uri golden) async {
+    golden = addPrefix(golden);
+    await update(golden, imageBytes);
     final File goldenFile = getGoldenFile(golden);
-    if (!goldenFile.existsSync()) {
-      throw TestFailure('Could not be compared against non-existent file: "$golden"');
-    }
-    final List<int> goldenBytes = await goldenFile.readAsBytes();
-    final ComparisonResult result = GoldenFileComparator.compareLists(imageBytes, goldenBytes);
-    return result.passed;
+
+    return await skiaClient.imgtestAdd(golden.path, goldenFile);
   }
 
-  /// Decides based on the current platform whether goldens tests should be
-  /// performed against the flutter/goldens repository.
-  static bool isAvailableOnPlatform(Platform platform) => platform.isLinux;
+  /// Decides based on the current environment whether goldens tests should be
+  /// performed against Skia Gold.
+  static bool isAvailableForEnvironment(Platform platform) {
+    final String cirrusCI = platform.environment['CIRRUS_CI'] ?? '';
+    final String cirrusPR = platform.environment['CIRRUS_PR'] ?? '';
+    final String cirrusBranch = platform.environment['CIRRUS_BRANCH'] ?? '';
+    final String goldServiceAccount = platform.environment['GOLD_SERVICE_ACCOUNT'] ?? '';
+    return cirrusCI.isNotEmpty
+      && cirrusPR.isEmpty
+      && cirrusBranch == 'master'
+      && goldServiceAccount.isNotEmpty;
+  }
 }
 
-/// A [FlutterGoldenFileComparator] for testing golden images with Skia Gold.
+/// A [FlutterGoldenFileComparator] for testing golden images before changes are
+/// merged into the master branch.
 ///
-/// For testing across all platforms, the [SkiaGoldClient] is used to upload
-/// images for framework-related golden tests and process results. Currently
-/// these tests are designed to be run post-submit on Cirrus CI, informed by the
-/// environment.
+/// This comparator utilizes the [SkiaGoldClient] to request baseline images for
+/// the given device under test for comparison. This comparator is only
+/// initialized during pre-submit testing on Cirrus CI.
 ///
 /// See also:
 ///
 ///  * [GoldenFileComparator], the abstract class that
 ///    [FlutterGoldenFileComparator] implements.
-///  * [FlutterGoldensRepositoryFileComparator], another
-///    [FlutterGoldenFileComparator] that tests golden images using the
-///    flutter/goldens repository.
-class FlutterSkiaGoldFileComparator extends FlutterGoldenFileComparator {
-  /// Creates a [FlutterSkiaGoldFileComparator] that will test golden file
-  /// images against Skia Gold.
+///  * [FlutterSkiaGoldFileComparator], another
```

Comment:
here (and elsewhere where these cross-links exists): the difference between the cooperators is not really made clear in these snippets.